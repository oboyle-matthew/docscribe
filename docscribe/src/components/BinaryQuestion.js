import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { CheckBox } from 'react-native-elements';
import AppStore from '../stores/AppStore';

export default class BinaryQuestion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      oneChecked: false,
      twoChecked: false
    };
  }

  checkOne = () => {
    this.props.app.updateFirebase(this.props.fb, true);
    const { oneChecked, twoChecked } = this.state;
    if (!oneChecked && twoChecked) {
      this.setState({
        twoChecked: false,
      });
    }
    this.setState({
      oneChecked: !oneChecked,
    });
  };

  checkTwo = () => {
    this.props.app.updateFirebase(this.props.fb, false);
    const { oneChecked, twoChecked } = this.state;
    if (!twoChecked && oneChecked) {
      this.setState({
        oneChecked: false,
      });
    }
    this.setState({
      twoChecked: !twoChecked,
    });
  };

  render() {
    const { question, optionOne, optionTwo } = this.props;
    const { oneChecked, twoChecked } = this.state;
    return (
      <View style={{ display: 'flex', flexDirection: 'row' }}>
        <Text style={{ width: '40%', textAlign: 'center' }}>{question}</Text>
        <View style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between' }}>
          <CheckBox title={optionOne} checked={oneChecked} onPress={this.checkOne} />
          <CheckBox title={optionTwo} checked={twoChecked} onPress={this.checkTwo} />
        </View>
      </View>
    );
  }
}

BinaryQuestion.propTypes = {
  question: PropTypes.string.isRequired,
  optionOne: PropTypes.string.isRequired,
  optionTwo: PropTypes.string.isRequired,
  fb: PropTypes.string.isRequired,
  app: PropTypes.instanceOf(AppStore).isRequired,
};
