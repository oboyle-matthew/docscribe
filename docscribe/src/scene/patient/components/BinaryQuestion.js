import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { CheckBox } from 'react-native-elements';

export default class BinaryQuestion extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            oneChecked: this.props.app.object[this.props.objectKey] === true,
            twoChecked: this.props.app.object[this.props.objectKey] === false,
        };
    }

    checkOne = () => {
        if (this.state.oneChecked) {
            this.props.app.object[this.props.objectKey] = null;
            this.props.onValueChanged && this.props.onValueChanged(null);
        } else {
            this.props.app.object[this.props.objectKey] = true;
            this.props.onValueChanged && this.props.onValueChanged(true);
        }
        const { oneChecked, twoChecked } = this.state;
        if (!oneChecked && twoChecked) {
            this.setState({
                twoChecked: false,
            });
        }
        this.setState({
            oneChecked: !oneChecked,
        });
    };

    checkTwo = () => {
        if (this.state.twoChecked) {
            this.props.app.object[this.props.objectKey] = null;
            this.props.onValueChanged && this.props.onValueChanged(null);
        } else {
            this.props.app.object[this.props.objectKey] = false;
            this.props.onValueChanged && this.props.onValueChanged(false);
        }
        const { oneChecked, twoChecked } = this.state;
        if (!twoChecked && oneChecked) {
            this.setState({
                oneChecked: false,
            });
        }
        this.setState({
            twoChecked: !twoChecked,
        });
    };

    render() {
        const { question, optionOne, optionTwo } = this.props;
        const { oneChecked, twoChecked } = this.state;
        return (
            <View style={{ display: 'flex', flexDirection: 'row'}}>
                <Text style={{ width: 100, textAlign: 'center' }}>{question}</Text>
                <View style={{display: 'flex', flexDirection: 'row', justifyContent: 'space-between' }}>
                    <CheckBox title={optionOne} checked={oneChecked} onPress={this.checkOne} />
                    <CheckBox title={optionTwo} checked={twoChecked} onPress={this.checkTwo} />
                </View>
            </View>
        );
    }
}

BinaryQuestion.propTypes = {
    question: PropTypes.string.isRequired,
    optionOne: PropTypes.string.isRequired,
    optionTwo: PropTypes.string.isRequired,
};
