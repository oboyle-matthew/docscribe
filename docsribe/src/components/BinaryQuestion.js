import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { CheckBox } from 'react-native-elements';

export default class BinaryQuestion extends React.Component {

    constructor() {
        super();
        this.state = {
            oneChecked: false,
            twoChecked: false
        }
    }

    checkOne = () => {
        if (!this.state.oneChecked && this.state.twoChecked) {
            this.setState({
                twoChecked: false
            })
        }
        this.setState({
            oneChecked: !this.state.oneChecked,
        })
    };

    checkTwo = () => {
        if (!this.state.twoChecked && this.state.oneChecked) {
            this.setState({
                oneChecked: false
            })
        }
        this.setState({
            twoChecked: !this.state.twoChecked,
        })
    };

    render() {
        const { question } = this.props;
        return (
            <View style={{display: 'flex', flexDirection: 'row'}}>
                <Text style={{width: '40%', textAlign: 'center'}}>{question}</Text>
                <View style={{display: 'flex', flexDirection: 'row', justifyContent: 'space-between'}}>
                    <CheckBox
                        title={this.props.optionOne}
                        checked={this.state.oneChecked}
                        onPress={this.checkOne}
                    />
                    <CheckBox
                        title={this.props.optionTwo}
                        checked={this.state.twoChecked}
                        onPress={this.checkTwo}
                    />
                </View>

            </View>
        );
    }
}

BinaryQuestion.propTypes = {
    question: PropTypes.string
}