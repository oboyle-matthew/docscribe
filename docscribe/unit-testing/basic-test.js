import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BinaryQuestion from '../src/components/BinaryQuestion';

// Basic test made! (Try changing option two to "maybe" and it will fail!)
it('renders correctly', () => {
  const pain = renderer
    .create(
      <BinaryQuestion
        question="Did you adhere to the prescribed usage guidelines today?"
        optionOne="Yes"
        optionTwo="No"
        style={{ marginBottom: '20%' }}
      />
    )
    .toJSON();
  expect(pain).toMatchSnapshot();
});
