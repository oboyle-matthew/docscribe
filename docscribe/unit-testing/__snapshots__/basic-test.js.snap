// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "textAlign": "center",
        "width": "40%",
      }
    }
  >
    Did you adhere to the prescribed usage guidelines today?
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          No
        </Text>
      </View>
    </View>
  </View>
</View>
`;
